.github/workflows/dependabot-code-review.yml

```
@@ -0,0 +1,39 @@
+name: Dependabot
+on:
+  pull_request:
+    # types:
+    #   - opened
+    branches:
+      - '**'
+
+permissions:
+  pull-requests: write
+  checks: read
+
+jobs:
+  run:
+    name: Request Code Review & Approve PR if checks pass
+    runs-on: ubuntu-latest
+    # if: github.event.pull_request.user.login == 'dependabot[bot]'
+    steps:
+      - name: Request Code Review on Slack
+        uses: slackapi/slack-github-action@v2.1.0
+        with:
+          webhook: ${{ secrets.SLACK_WEBHOOK_URL }}
+          webhook-type: incoming-webhook
+          payload: |
+            "text": "Please review ${{ github.event.pull_request.html_url }}"
+
+      - name: Ensure all checks passing before auto approval
+        uses: poseidon/wait-for-status-checks@v0.6.0
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+          delay: 300s
```

Wait for 5 minutes before checking, as some of our tests take 10+ minutes to complete. If checks fail early due to build errors, it means that we don't have to wait for too long for checks to complete. Alternatively, I could bump this up to 10mins to cater for the long running cypress tests?

---

Don't see what the harm in waiting for the Cypress tests to pass is in this case. Better to only approve when everything is passing

---

Why don't I increase the delay to 12 mins, so that it won't be waiting for any of the checks to run (as they should be completed by then), and will only proceed to the next stage if all of those checks have passed.